test/test_videoapi.py

```diff
@@ -0,0 +1,213 @@
+    ),
+    "WUzgd7C1pWA.mp4": GroundTruth(
+        duration=11.0, video_fps=29.97, audio_sample_rate=48000
+    ),
+}
+
+
+@unittest.skipIf(_HAS_VIDEO_OPT is False, "Didn't compile with ffmpeg")
+class TestVideoApi(unittest.TestCase):
+    def test_predefined_metadata(self):
+        """
+        Test that the source metadata corresponds to the one returned
+        by the new video decoder API.
+        """
+        for test_video, config in test_videos.items():
+            full_path = os.path.join(VIDEO_DIR, test_video)
+            reader = VideoReader(full_path, "video")
+            reader_md = reader.get_metadata()
+            self.assertAlmostEqual(
+                config.video_fps, reader_md["video"]["fps"][0], delta=0.0001
+            )
+            self.assertAlmostEqual(
+                config.duration, reader_md["video"]["duration"][0], delta=0.5
+            )
+
+    def test_frame_reading(self):
+        for test_video, config in test_videos.items():
+            full_path = os.path.join(VIDEO_DIR, test_video)
+
+            av_reader = av.open(full_path)
+
+            if av_reader.streams.video:
+                video_reader = VideoReader(full_path, "video")
+                for av_frame in av_reader.decode(av_reader.streams.video[0]):
+                    vr_frame = next(video_reader)
+
+                    self.assertAlmostEqual(
+                        float(av_frame.pts * av_frame.time_base),
+                        vr_frame["pts"],
+                        delta=0.1,
+                    )
+
+                    av_array = torch.tensor(av_frame.to_rgb().to_ndarray()).permute(
+                        2, 0, 1
+                    )
+                    vr_array = vr_frame["data"]
+                    mean_delta = torch.mean(
+                        torch.abs(av_array.float() - vr_array.float())
+                    )
+                    # on average the difference is very small and caused
+                    # by decoding
+                    # TODO: asses empirically how to set this?
+                    self.assertTrue(mean_delta.item() < 2.5)
+
+            av_reader = av.open(full_path)
+            if av_reader.streams.audio:
+                video_reader = VideoReader(full_path, "audio")
+                for av_frame in av_reader.decode(av_reader.streams.audio[0]):
+                    vr_frame = next(video_reader)
+                    self.assertAlmostEqual(
+                        float(av_frame.pts * av_frame.time_base),
+                        vr_frame["pts"],
+                        delta=0.1,
+                    )
+
+                    av_array = torch.tensor(av_frame.to_ndarray()).permute(1, 0)
+                    vr_array = vr_frame["data"]
+
+                    mean_delta = torch.mean(
+                        torch.abs(av_array.float() - vr_array.float())
+                    )
+                    self.assertTrue(mean_delta.item() < 1)
```

Comment:
What is the range of values for the audio data?

Reply:
[-1, 1]

Reply:
So this assert doesn't actually compare the values of the audio data, given that the tolerance is so large?